lib/_stream_writable.js

```diff
@@ -429,13 +429,7 @@ function onwrite(stream, er) {
       onwriteError(stream, state, er, cb);
     }
   } else {
-    // Check if we're actually ready to finish, but don't emit yet
-    const finished = needFinish(state) || stream.destroyed;
-
-    if (!finished &&
-        !state.corked &&
-        !state.bufferProcessing &&
-        state.bufferedRequest) {
+    if (state.bufferedRequest) {
```

Comment:
This condition is a duplicate of what's inside of `clearBuffer`. However, it is here on purpose since `clearBuffer` is not inlineable.